ledger/ledger_test.go

```diff
@@ -1078,3 +1078,137 @@ func TestLedgerReload(t *testing.T) {
+func generateCreatables(numElementsPerSegement int) (
+	randomCtbs map[basics.CreatableIndex]modifiedCreatable,
+	assetID3,
+	appID3 basics.CreatableIndex,
+	err error) {
+	_, randomCtbs = randomCreatables(numElementsPerSegement)
+	asCounter3 := 0
+	apCounter3 := 0
+
+	for x := 0; x < 10; x++ {
+		// find the assetid greater than at least 2 assetids
+		for cID, crtble := range randomCtbs {
+			switch crtble.ctype {
+			case basics.AssetCreatable:
+				if assetID3 == 0 {
+					assetID3 = cID
+					continue
+				}
+				asCounter3++
+				if assetID3 < cID {
+					assetID3 = cID
+				}
+			case basics.AppCreatable:
+				if appID3 == 0 {
+					appID3 = cID
+					continue
+				}
+				apCounter3++
+				if appID3 < cID {
+					appID3 = cID
+				}
+			}
+			if asCounter3 >= 3 && apCounter3 >= 3 {
+				// found at least 3rd smallest of both
+				break
+			}
+		}
+
+		// there should be at least 3 asset and 3 app creatables generated.
+		// In the rare event this does not happen, repeat... up to 10 times (x)
+		if asCounter3 >= 3 && apCounter3 >= 3 {
+			break
+		}
+	}
+	if asCounter3 < 3 && apCounter3 < 3 {
+		return nil, 0, 0, fmt.Errorf("could not generate 3 apps and 3 assets")
+	}
+	return
+}
+
+// TestListAssetsAndApplications tests the ledger.ListAssets and ledger.ListApplications
+// interfaces. The detailed test on the correctness of these functions is given in:
+// TestListCreatables (acctupdates_test.go)
+func TestListAssetsAndApplications(t *testing.T) {
+
+	numElementsPerSegement := 10 // This is multiplied by 10. see randomCreatables
+
+	//initLedger
+	genesisInitState, _ := testGenerateInitState(t, protocol.ConsensusCurrentVersion)
+	const inMem = true
+	log := logging.TestingLog(t)
+	cfg := config.GetDefaultLocal()
+	cfg.Archival = true
+	ledger, err := OpenLedger(log, t.Name(), inMem, genesisInitState, cfg)
+	require.NoError(t, err, "could not open ledger")
+	defer ledger.Close()
+
+	// ******* All results are obtained from the cache. Empty database *******
+	// ******* No deletes                                              *******
+	// get random data. Inital batch, no deletes
+	randomCtbs, maxAsset, maxApp, err := generateCreatables(numElementsPerSegement)
+	require.NoError(t, err)
+
+	// set the cache
+	ledger.accts.creatables = randomCtbs
+
+	// Test ListAssets
+	// Check the number of results limit
+	results, err := ledger.ListAssets(basics.AssetIndex(maxAsset), 2)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(results))
+	// Check the max asset id limit
+	results, err = ledger.ListAssets(basics.AssetIndex(maxAsset), 100)
+	assetCount := 0
+	for id, ctb := range randomCtbs {
+		if ctb.ctype == basics.AssetCreatable &&
+			ctb.created &&
+			id <= maxAsset {
+			assetCount++
+		}
+	}
+	require.Equal(t, assetCount, len(results))
+
+	// Test ListApplications
+	// Check the number of results limit
+	ledger.accts.creatables = randomCtbs
+	results, err = ledger.ListApplications(basics.AppIndex(maxApp), 2)
```

Comment:
you should do something with the `results` variable here. Maybe test it's size like you did with the assets above ?
if that's not applied to this function, then just do `_, err = ledger.ListApplication .... `

Reply:
Yep. forgot the test the count here.